test/ngAnimate/integrationSpec.js

```diff
@@ -25,6 +25,32 @@ describe('ngAnimate integration tests', function() {
     ss.destroy();
   });
 
+ it('should cancel a running and started removeClass animation when a follow-up addClass animation adds the same class',
```

Comment:
Yes, another very good question. Internally, there's a running state, but that can mean two things: first, the animation process waits for the next digest to happen; then it flushes an requestAnimationFrame (for the reflow) to trigger the actual animation start. The previous test didn'T flush the rAF, and so the scenario wasn't created.